### DISCUSSION TITLE
Is it possible to create firmware for Ender 3 S1 with F4 chip to support both LA and CV Laser?

### DISCUSSION THREAD

**Discussion by mrigne** [mriscoc/Ender3V2S1#932] [Q&A] [answered]
Is it possible to create firmware for Ender 3 S1 with F4 chip to support both LA and CV Laser? 
Currently I have LA version, but sometimes I'd like to use Creality laser module. 
Does this machine can run version woth both features?

**Comment by mriscoc** [comment, marked as answer]
The different chips used on the various Creality boards have different amounts of flash for firmware storage, this means that different boards will be limited by flash size to which firmware features can be enabled while still allowing the firmware to fit. The F4 chip has the smallest amount of usable flash. The maximum firmware size for the F4 is 192k, any compiled firmwares over 192k will not work due to the mentioned size limitations, the precompiled firmwares for the F4 boards have some minor features removed to allow for stuff like UBL, LA, CVLM and other new marlin features. https://github.com/mriscoc/Ender3V2S1/issues/648

So, could be very difficult to make space to have LA and CVLM at the same time.

**Reply by nicholasstparker** [reply]
I really really would like LA and UBL at the same time on the F4.. Could I remove the filament runout sensor? I'm so bummed I don't have the F1.

**Reply by mriscoc** [reply]
Yes, you could disable all code related to the filament runout sensor.

**Reply by nicholasstparker** [reply]
Thank you! I was able to compile a version for the F4 with UBL, MPC and LA and it came in at 192kb. I was able to keep the filament sensor but disabled things like long name file scrolling, arc support, no volumetrics, show bootscreen and a few others.

**Comment by mrigne** [comment]
Thank you for your response. Some magic (or not) was done and i was able to build firmware that fit flash. I have disabled theme colors, gcode preview and lockscreen, also switched from MPC back to PID and result firmware fit progmem. Tested both laser and la - all works like a charm! Thank you very much for configurator - now I’m absolutely happy. 
it would be grat if it will be possible to disable filament runout sensor btw - i home that it will allow to use MPC again :)

**Reply by mriscoc** [reply]
You can write your own json with commands to disable the features that you don't want and add that json file in the _features folder, so you can select it in the future.

**Comment by mrigne** [comment]
Also removed UBL support - right now benchy is printing and it was enough to use bilinear
